### PR TITLE
Small review on docs & code structure in JControllerLegacy library classes

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes
- Improves doc blocks
- Simplify some lines as able
- Access input object from within the controller instance versus JFactory
#### Testing Instructions

Review

### CODE REVIEW

File: libraries/legacy/controller/form.php [hunk 1]
```diff
@@ -353,15 +350,14 @@ public function cancel($key = null)
 	 *
 	 * @param   string  $key     The name of the primary key of the URL variable.
 	 * @param   string  $urlVar  The name of the URL variable if different from the primary key
-	 * (sometimes required to avoid router collisions).
+	 *                           (sometimes required to avoid router collisions).
```
Comment (brianteeman) [inline libraries/legacy/controller/form.php]: Is this supposed to be indebted ?

Comment (mbabker) [inline libraries/legacy/controller/form.php]: Multiline comments like this are, yes.  See JHtml as the main example.

Comment (brianteeman) [inline libraries/legacy/controller/form.php]: Oops didn't spot it was a continuation of the previous line. Sorry to bother you

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on af6f324464834f607fba98d09b4f79128662a9e5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11058">issues.joomla.org/joomla-cms/11058</a>.</sub>

**Comment by tomartailored** [comment]
I have tested this item :white_check_mark: successfully on af6f324464834f607fba98d09b4f79128662a9e5<br /><br />I have Test this item successfully and code is optimised<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11058">issues.joomla.org/joomla-cms/11058</a>.</sub>

**Pull request merged**